plugins/modules/azure_rm_afdendpoint.py

```diff
@@ -0,0 +1,351 @@
+        name=endpoint.name,
+        provisioning_state=endpoint.provisioning_state,
+        tags=endpoint.tags,
+        type=endpoint.type
+    )
+
+
+class AzureRMEndpoint(AzureRMModuleBase):
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            auto_generated_domain_name_label_scope=dict(
+                type='str',
+                default = 'TenantReuse',
+                choices=["TenantReuse", "SubscriptionReuse", "ResourceGroupReuse", "NoReuse"]
+            ),
+            enabled_state=dict(
+                type='str',
+                default = 'Enabled',
+                choices=['Enabled', 'Disabled']
+            ),
+            location=dict(
+                type='str'
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            profile_name=dict(
+                type='str',
+                required=True
+            ),
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+        
+        self.auto_generated_domain_name_label_scope = None
+        self.enabled_state = None
+        self.location = None
+        self.tags = None
+
+        self.name = None
+        self.profile_name = None
+        self.resource_group = None
+        self.state = None
+
+        self.endpoint_client = None
+
+        self.results = dict(changed=False)
+
+        super(AzureRMEndpoint, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                supports_check_mode=True,
+                                                supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.endpoint_client = self.get_endpoint_client()
+
+        to_be_updated = False
+
+        if not self.location:
+            resource_group = self.get_resource_group(self.resource_group)
+            self.location = resource_group.location
+
+        response = self.get_endpoint()
+
+        if self.state == 'present':
+
+            if not response:
+                self.log("Need to create the AFD Endpoint")
+
+                if not self.check_mode:
+                    new_response = self.create_endpoint()
+                    self.results['id'] = new_response['id']
+                    self.results['host_name'] = new_response['host_name']
+                    self.log("AFD Endpoint creation done")
+
+                self.results['changed'] = True
+                return self.results
+            
+            else:
+                self.log('Results : {0}'.format(response))
+                self.results['id'] = response['id']                
```

Comment:
```suggestion
                self.results['id'] = response['id']
```

Reply:
Not sure I understand this one, as there is no response in this section of code, so setting self.results['id'] = response['id'] would be blank.  I set it to new_response['id'] as the return result from the create_endpoint() function.  I could honestly be missing something though..  :)

Reply:
Or are you referring to having a value when in check mode?